### PR TITLE
Implement t12 procs

### PR DESCRIPTION
**Pull request by zoristaken**
## **1. Raid items**

- [x] [Vessel of Acceleration](https://www.wowhead.com/cata/item=68995/vessel-of-acceleration)

- [x] [Vessel of Acceleration Heroic](https://www.wowhead.com/cata/item=69167/vessel-of-acceleration)

- [x] [Matrix Restabilizer](https://www.wowhead.com/cata/item=68994/matrix-restabilizer)

- [x] [Matrix Restabilizer Heroic](https://www.wowhead.com/cata/item=69150/matrix-restabilizer)

- [x] [Variable Pulse Lightning Capacitor](https://www.wowhead.com/cata/item=68925/variable-pulse-lightning-capacitor)

- [x] [Variable Pulse Lightning Capacitor Heroic](https://www.wowhead.com/cata/item=69110/variable-pulse-lightning-capacitor)

- [ ] [Jaws of Defeat](https://www.wowhead.com/cata/item=68926/jaws-of-defeat)

- [ ] [Jaws of Defeat Heroic](https://www.wowhead.com/cata/item=69111/jaws-of-defeat)

- [x] [The Hungerer](https://www.wowhead.com/cata/item=68927/the-hungerer)

- [x] [The Hungerer Heroic](https://www.wowhead.com/cata/item=69112/the-hungerer)

- [ ] [Spidersilk Spindle](https://www.wowhead.com/cata/item=68981/spidersilk-spindle)

- [ ] [Spidersilk Spindle Heroic](https://www.wowhead.com/cata/item=69138/spidersilk-spindle)

- [x] [Necromatic Focus](https://www.wowhead.com/cata/item=68982/necromantic-focus)

- [x] [Necromatic Focus Heroic](https://www.wowhead.com/cata/item=69139/necromantic-focus)

- [ ] [Eye of Blazing Power](https://www.wowhead.com/cata/item=68983/eye-of-blazing-power)

- [ ] [Eye of Blazing Power Heroic](https://www.wowhead.com/cata/item=69149/eye-of-blazing-power)

- [x] [Apparatus of Khaz'goroth](https://www.wowhead.com/cata/item=68972/apparatus-of-khazgoroth)

- [x] [Apparatus of Khaz'goroth Heroic](https://www.wowhead.com/cata/item=69113/apparatus-of-khazgoroth)

- [ ] [Scales of Life](https://www.wowhead.com/cata/item=68915/scales-of-life)

- [ ] [Scales of Life Heroic](https://www.wowhead.com/cata/item=69109/scales-of-life) 

## **2. Avengers of Hyjal and Quest Rewards**
 There seems to be heroic versions of the avengers of hyjal items in the database, but those were not available back in the day, just like the heroic trash loot from tier 11 raids.
- [ ] [Stay of Execution](https://www.wowhead.com/cata/item=68996/stay-of-execution)

- [x] [Rune of Zeth](https://www.wowhead.com/cata/item=68998/rune-of-zeth)

- [x] [Fiery Quintessence](https://www.wowhead.com/cata/item=69000/fiery-quintessence)

- [x] [Ancient Petrified Seed](https://www.wowhead.com/cata/item=69001/ancient-petrified-seed)

- [x] [Essence of the Eternal Flame](https://www.wowhead.com/cata/item=69002/essence-of-the-eternal-flame) 

- [x] [Dwyer's Caber](https://www.wowhead.com/cata/item=70141/dwyers-caber)

- [x] [Moonwell Chalice](https://www.wowhead.com/cata/item=70142/moonwell-chalice)

- [x] [Moonwell Phial](https://www.wowhead.com/cata/item=70143/moonwell-phial)

- [x] [Ricket's Magnetic Fireball](https://www.wowhead.com/cata/item=70144/rickets-magnetic-fireball)

## Brewfest
 Do we even bother implementing the tank's brewmaiden on use? [bubblier](https://www.wowhead.com/cata/item=71334/bubblier-brightbrew-charm) and [bitterer](https://www.wowhead.com/cata/item=71333/bitterer-balebrew-charm)
- [x] [Brawler's Trophy](https://www.wowhead.com/cata/item=71338/brawlers-trophy)

- [x] [Coren's Chilled Chromium Coaster](https://www.wowhead.com/cata/item=71335/corens-chilled-chromium-coaster)

- [x] [Mithril Stopwatch](https://www.wowhead.com/cata/item=71337/mithril-stopwatch)

- [x] [Petrified Pickled Egg](https://www.wowhead.com/cata/item=71336/petrified-pickled-egg)

## TBD
The legendary staff will require more thought than any of the other items to implement. There needs to be a discussion if we even bother implementing it following simc and old data for now, or we wait for more information.
- [ ] [Dragonwrath, Tarecgosa's Rest](https://www.wowhead.com/cata/item=71086/dragonwrath-tarecgosas-rest)

### CODE REVIEW

**Comment by zoristaken** [comment]
Issue #806 will help implementing the Matrix Restabilizer and Apparatus of Khaz'goroth trinkets

**Comment by zoristaken** [comment]
Some additional comments, the trinkets that have not been implemented yet are either healing trinkets that I don't know how to go about, or have some sort of mechanic that I didnt investigate if the core allows it right now (example: stacking mana cost reduction during aura on Jaws of Defeat).

I also have some TODO comments that need addressing. Need to double check that I didn't mess up any numbers but otherwise would like a review of what's already implemented

File: sim/common/cata/damage_procs.go [hunk 1]
```diff
@@ -39,4 +41,95 @@ func init() {
 			Callback: core.CallbackOnSpellHitDealt,
 		},
 	})
+
+	core.NewItemEffect(68925, func(agent core.Agent) {
+		character := agent.GetCharacter()
+		procChance := 0.3 //TODO: placeholder, get real proc chance
```
Comment (hillerstorm) [inline sim/common/cata/damage_procs.go:47]: simc (latest version before MoP) has these at 15% chance, should probably start there and see if anything needs adjusting after launch.

File: sim/common/cata/stat_bonus_procs.go [hunk 2]
```diff
@@ -869,6 +869,84 @@ func init() {
...
+	shared.NewProcStatBonusEffect(shared.ProcStatBonusEffect{
+		Name:       "The Hungerer",
+		ID:         68927,
+		AuraID:     96911,
+		Bonus:      stats.Stats{stats.MeleeHaste: 1532, stats.SpellHaste: 1532},
+		Duration:   time.Second * 15,
+		Callback:   core.CallbackOnSpellHitDealt,
+		ProcMask:   core.ProcMaskMeleeOrRanged,
+		Outcome:    core.OutcomeLanded,
+		ProcChance: 1, //TODO: verify proc chance, seems wrong?
```
Comment (hillerstorm) [inline sim/common/cata/stat_bonus_procs.go:934]: simc also has this at 100%, seems weird but the 60s ICD makes it fair I guess

**Comment by hillerstorm** [comment]
Tank trinkets should be pretty straight forward to implement, at least by someone with a bit of absorb-knowledge (e.g. @rosenrusinov 😅).

[Jaws of Defeat](https://www.wowhead.com/cata/item=68926/jaws-of-defeat) should be possible to implement with a dynamic spell mod using SpellMod_PowerCost_Flat right? Just reset the FloatValue on gain and add 110 to the current each stack?

File: sim/common/cata/damage_procs.go (hunk 1 above)
Comment (zoristaken) [reply sim/common/cata/damage_procs.go:47]: I will update these to simc value, I tried to find the proc chance in the db/wowhead but had no success finding it. Think this is good for now.

File: sim/common/cata/stat_bonus_procs.go (hunk 2 above)
Comment (zoristaken) [reply sim/common/cata/stat_bonus_procs.go:934]: Yeah I double checked the db and it is indeed 100%, I think this trinket received a rework from its original version and might be why it's different

**Comment by zoristaken** [comment]
> Tank trinkets should be pretty straight forward to implement, at least by someone with a bit of absorb-knowledge (e.g. @rosenrusinov 😅).
> 
> [Jaws of Defeat](https://www.wowhead.com/cata/item=68926/jaws-of-defeat) should be possible to implement with a dynamic spell mod using SpellMod_PowerCost_Flat right? Just reset the FloatValue on gain and add 110 to the current each stack?

I might leave tank/healing trinkets not implemented by now and we can just focus and making sure the ones I have implemented are correct. If that's the case I will just open an issue to track what's left to implement

**Comment by 1337LutZ** [comment]
> > Tank trinkets should be pretty straight forward to implement, at least by someone with a bit of absorb-knowledge (e.g. @rosenrusinov 😅).
> > [Jaws of Defeat](https://www.wowhead.com/cata/item=68926/jaws-of-defeat) should be possible to implement with a dynamic spell mod using SpellMod_PowerCost_Flat right? Just reset the FloatValue on gain and add 110 to the current each stack?
> 
> I might leave tank/healing trinkets not implemented by now and we can just focus and making sure the ones I have are correct. If that's the case I will just open an issue to track what's left to implement

@zoristaken I tend to agree. There's more questions about P2 on a lot of the class Discords, so having the DPS trinkets/procs implemented for now would already help and let theorycrafters start cooking.

**Comment by rosenrusinov** [comment]
I agree with what was mentioned above and we can merge this as it is and I’ll work on the tank trinkets when I’m back from holidays

**Pull request merged**